test/ReverseProxy.FunctionalTests/WebSocketTests.cs

```diff
@@ -0,0 +1,239 @@
+using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.WebSockets;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.ReverseProxy.Abstractions;
+using Microsoft.ReverseProxy.Configuration;
+using Xunit;
+
+namespace Microsoft.ReverseProxy
+{
+    public class WebSocketTests
+    {
+        [Theory]
+        [InlineData(WebSocketMessageType.Binary)]
+        [InlineData(WebSocketMessageType.Text)]
+        public async Task WebSocketTest(WebSocketMessageType messageType)
+        {
+            var destinationHost = CreateWebSocketHost();
```

Comment:
Stop and Dispose all the hosts, clients, sockets, streams, etc. by the end of the test.
```suggestion
            using var destinationHost = CreateWebSocketHost();
```